### ISSUE TITLE
fix(link-fragments): slugGitea must not include user-content- prefix

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#219]
## Overview

`slugGitea` incorrectly includes a `user-content-` prefix in the returned slug. Users write fragment links without this prefix, so every valid cross-heading link is flagged as broken.

## Problem

Gitea (and Forgejo) add `user-content-` to the `id` attribute of heading elements in the rendered DOM for CSP isolation. However, the `href` on the auto-generated anchor links — which matches what users write in Markdown source — does **not** include the prefix:

```html
<!-- DOM -->
<h2 id="user-content-what-does-forgejo-offer">...</h2>
<a class="anchor" href="#what-does-forgejo-offer">¶</a>
```

A Markdown link `[link](#what-does-forgejo-offer)` is valid and works. The `user-content-` prefix is a rendering implementation detail, not part of the fragment that users write.

This is the same pattern used by GitHub and GitLab.

### Verified on

- Forgejo: `codeberg.org/forgejo/forgejo` (Playwright scrape)
- Gitea: `gitea.com/gitea/go-sdk` (Playwright scrape)

### Impact

```
Heading:             ## What does Forgejo offer?
User writes:         [link](#what-does-forgejo-offer)         ← valid
slugGitea produces:  user-content-what-does-forgejo-offer     ← never matches → false positive
```

## Fix

Remove the prefix from `slugGitea`:

```go
// slugGitea computes the Gitea (goldmark) slug.
// Algorithm is identical to GitHub (github-slugger v2).
// Note: Gitea adds "user-content-" to the DOM id attribute for CSP isolation,
// but users write fragment links without that prefix.
func slugGitea(text string) string {
    return slugGitHub(text)
}
```

`forgejo` already delegates to `slugGitea`, so it is fixed automatically.

The emoji-stripping bug in `slugGitHub` (tracked in #211) also applies here but is a separate issue.

## References

- `internal/rule/link_fragments_slug.go` — `slugGitea`
- Verification comment: https://github.com/shinagawa-web/gomarklint/issues/206#issuecomment-4341850596
- Tracking issue: #206

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create
- none

### Files to modify
- `internal/rule/link_fragments_slug.go` — change `slugGitea` to return `slugGitHub(text)` without `"user-content-"` prefix; update doc comment
- `internal/rule/link_fragments_slug_test.go` — update/add test vectors for `gitea` and `forgejo` presets without prefix

### Checklist
- [ ] Fix `slugGitea` in `link_fragments_slug.go`
- [ ] Update unit tests in `link_fragments_slug_test.go`
- [ ] `make test-all`
